lib/services/search_service.dart

```diff
@@ -327,4 +368,52 @@ class SearchService {
         location.longitude < locationData.bbox[2] &&
         location.latitude < locationData.bbox[3];
   }
+
+  List<Tuple3<int, MonthData, int>> _getPossibleEventDate(String query) {
+    final List<Tuple3<int, MonthData, int>> possibleEvents = [];
+    if (query.trim().isEmpty) {
+      return possibleEvents;
+    }
+    final result = query
+        .trim()
+        .split(RegExp('[ ,-/]+'))
+        .map((e) => e.trim())
+        .where((e) => e.isNotEmpty)
+        .toList();
+    final resultCount = result.length;
+    if (resultCount < 1 || resultCount > 4) {
+      return possibleEvents;
+    }
+
+    final int day = int.tryParse(result[0]);
+    if (day == null || day < 1 || day > 31) {
+      return possibleEvents;
+    }
+    final List<MonthData> potentialMonth =
+        resultCount > 1 ? _getMatchingMonths(result[1]) : allMonths;
+    final int parsedYear = resultCount >= 3 ? int.tryParse(result[2]) : null;
+    final List<int> matchingYears = [];
+    if (parsedYear != null) {
+      bool foundMatch = false;
+      for (int i = searchStartYear; i <= currentYear; i++) {
+        if (i.toString().startsWith(parsedYear.toString())) {
+          matchingYears.add(i);
+          foundMatch = foundMatch || (i == parsedYear);
+        }
+      }
+      if (!foundMatch && parsedYear > 1000 && parsedYear <= currentYear) {
```

Comment:
Could this be changed to >= searchStartYear ?

Reply:
Can change..
Just thinking that for manual input, may be we should not add any limitation on the searchStartYear.  For partial matches, it make sense to keep the number of years and relevant query param/durations list to reasonable size.

People can have photos (scanned version of print) older than 1970 :)